pkg/prometheus/promcfg.go

```diff
@@ -2099,6 +2100,15 @@ func (cg *ConfigGenerator) GenerateAgentConfiguration(
 		cfg = append(cfg, cg.generateRemoteWriteConfig(store))
 	}
 
+	if tracingConfig != nil {
+		tracingcfg, err := cg.generateTracingConfig(tracingConfig)
```

Comment:
`func (cg *ConfigGenerator) generateTracingConfig(tracingConfig *monitoringv1.PrometheusTracingConfig)`  signature could be changed to `func (cg *ConfigGenerator) generateTracingConfig()` since `cg.prom` holds the tracing config.

Reply:
Updated.. Can you check the new code @simonpasquier thanks :)